homeassistant/components/ccm15/climate.py

```diff
@@ -0,0 +1,166 @@
+        self._attr_temperature_unit = UnitOfTemperature.CELSIUS
+        self._attr_has_entity_name = True
+        self._attr_name = f"{self._ac_index}"
+        self._attr_target_temperature_step = 1
+        self._attr_hvac_modes = [
+            HVACMode.OFF,
+            HVACMode.HEAT,
+            HVACMode.COOL,
+            HVACMode.DRY,
+            HVACMode.AUTO,
+        ]
+        self._attr_fan_modes = [FAN_AUTO, FAN_LOW, FAN_MEDIUM, FAN_HIGH]
+        self._attr_swing_modes = [SWING_OFF, SWING_ON]
+        self._attr_supported_features = (
+            ClimateEntityFeature.TARGET_TEMPERATURE
+            | ClimateEntityFeature.FAN_MODE
+            | ClimateEntityFeature.SWING_MODE
+        )
+
+    @property
+    def unique_id(self) -> str:
+        """Return unique id."""
+        return f"{self._ac_host}.{self._ac_index}"
+
+    @property
+    def current_temperature(self) -> int | None:
+        """Return current temperature."""
+        if data := self.coordinator.get_ac_data(self._ac_index):
+            _LOGGER.debug("temp[%s]=%s", self._ac_index, data.temperature)
+            return data.temperature
+        return None
+
+    @property
+    def target_temperature(self) -> int | None:
+        """Return target temperature."""
+        if data := self.coordinator.get_ac_data(self._ac_index):
+            _LOGGER.debug("set_temp[%s]=%s", self._ac_index, data.temperature_setpoint)
+            return data.temperature_setpoint
+        return None
+
+    @property
+    def hvac_mode(self) -> HVACMode | None:
+        """Return hvac mode."""
+        if data := self.coordinator.get_ac_data(self._ac_index):
+            mode = data.ac_mode
+            _LOGGER.debug("hvac_mode[%s]=%s", self._ac_index, mode)
+            return CONST_CMD_STATE_MAP[mode]
+        return None
+
+    @property
+    def fan_mode(self) -> str | None:
+        """Return fan mode."""
+        if data := self.coordinator.get_ac_data(self._ac_index):
+            mode = data.fan_mode
+            _LOGGER.debug("fan_mode[%s]=%s", self._ac_index, mode)
+            return CONST_CMD_FAN_MAP[mode]
+        return None
+
+    @property
+    def swing_mode(self) -> str | None:
+        """Return swing mode."""
+        if data := self.coordinator.get_ac_data(self._ac_index):
+            _LOGGER.debug("is_swing_on[%s]=%s", self._ac_index, data.is_swing_on)
+            return SWING_ON if data.is_swing_on else SWING_OFF
+        return None
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return the device info."""
+        return DeviceInfo(
+            identifiers={
+                # Serial numbers are unique identifiers within a specific domain
+                (DOMAIN, f"{self._ac_host}.{self._ac_index}"),
+            },
+            name=self._attr_name,
+            manufacturer="Midea",
+            model="CCM15",
+        )
+
+    @property
+    def extra_state_attributes(self) -> dict[str, Any]:
```

Comment:
What happens when there is an error code?

Reply:
It will displayed as an extra attibutted.

Reply:
But what kind of errors can there be? Do we still get data? Can you maybe elaborate a bit more?

Reply:
In a ccm15 system there is a master device and slave devices (each in each room of a home). 
The master can be set for cooling or for heating. But there cannot be slaves that ask for heat and other for cool, that will be an error that the master will display.
The same error code will be displayed in the device in the room to the end user.

Reply:
Is this possible to cause without HA? Should we validate on our side if the device is able to go in a differnet mode?

Reply:
No, this is vendor specific.